pkg/forwarder/sync_forwarder.go

```diff
@@ -0,0 +1,138 @@
+package forwarder
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	utilhttp "github.com/DataDog/datadog-agent/pkg/util/http"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// SyncDefaultForwarder is a very simple Forwarder synchronously sending
+// the data to the intake.
+// It doesn't ship any retry mechanism for now.
+type SyncDefaultForwarder struct {
+	defaultForwarder *DefaultForwarder
+	client           *http.Client
+}
+
+// NewSyncDefaultForwarder returns a new synchronous forwarder.
+func NewSyncDefaultForwarder(keysPerDomains map[string][]string, timeout time.Duration) *SyncDefaultForwarder {
```

Comment:
Why `Default` ? Do  we have other SyncForwarders or plan on having them ?

Reply:
The naming isn't that great here but it is because it is a "synchronous implementation of the `DefaultForwarder`" -> `SyncDefaultForwarder`.
Do you think `SyncForwarder` would make more sense?

Reply:
I think so yes. `DefaultForwarder` was chose in case we wanted to do things like `SyncForwarder`.